backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/shipmentschedule/api/impl/HUShipmentScheduleBL.java

```diff
@@ -704,16 +705,20 @@ private void updateHUQuantitiesFromOrderLine(
 		final I_M_HU_LUTU_Configuration lutuConfiguration = //
 				deriveM_HU_LUTU_Configuration(shipmentSchedule);
 
-		//
-		// extract the packing material from lutu configuration
-		final HuPackingInstructionsVersionId versionId = HuPackingInstructionsVersionId.ofRepoId(lutuConfiguration.getM_LU_HU_PI_Item().getM_HU_PI_Version_ID());
-		final I_M_HU_PackingMaterial packingMaterial = handlingUnitsDAO.retrievePackingMaterialByPIVersionID(versionId, BPartnerId.ofRepoId(shipmentSchedule.getC_BPartner_ID()));
-
-		if (Objects.nonNull(packingMaterial)
-				&& packingMaterial.isQtyLUByMaxLoadWeight())
+		final I_M_HU_PI_Item m_lu_hu_pi_item = lutuConfiguration.getM_LU_HU_PI_Item();
+		if(Objects.nonNull(m_lu_hu_pi_item))
 		{
-			final BigDecimal qtyOrderedLU = lutuConfigurationFactory.calculateQtyLUForTotalQtyTUsByMaxWeight(lutuConfiguration, qtyTU_Effective, packingMaterial);
-			shipmentSchedule.setQtyOrdered_LU(qtyOrderedLU);
+			//
+			// extract the packing material from lutu configuration
+			final HuPackingInstructionsVersionId versionId = HuPackingInstructionsVersionId.ofRepoId(m_lu_hu_pi_item.getM_HU_PI_Version_ID());
+			final I_M_HU_PackingMaterial packingMaterial = handlingUnitsDAO.retrievePackingMaterialByPIVersionID(versionId, BPartnerId.ofRepoId(shipmentSchedule.getC_BPartner_ID()));
+
+			if (Objects.nonNull(packingMaterial)
+					&& packingMaterial.isQtyLUByMaxLoadWeight())
+			{
+				final BigDecimal qtyOrderedLU = lutuConfigurationFactory.calculateQtyLUForTotalQtyTUsByMaxWeight(lutuConfiguration, qtyTU_Effective, packingMaterial);
+				shipmentSchedule.setQtyOrdered_LU(qtyOrderedLU);
+			}
```

Comment:
I don't understand the code very well, but don't we need to handle the case that packingMaterial is null or isQtyLUByMaxLoadWeight() is false?
If not, please add a commend to avoid future FUD

Reply:
packingMaterial is null case was handled, and a proper comment was introduced too